From de38f706e226cfbefe2246a0b578a8f76e9c3e17 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 5 Nov 2012 11:19:19 +0000
Subject: [PATCH 109/118] RHEL 6: Ignore /etc/release if /etc/redhat-release
 exists (RHBZ#872454).

If the user has created /etc/release on a RHEL machine, don't
mis-detect the OS as NetBSD.
---
 src/inspect-fs.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/inspect-fs.c b/src/inspect-fs.c
index 276e7ef..cdfc089 100644
--- a/src/inspect-fs.c
+++ b/src/inspect-fs.c
@@ -209,7 +209,8 @@ check_filesystem (guestfs_h *g, const char *device,
   else if (is_dir_etc &&
            is_dir_bin &&
            guestfs_is_file (g, "/etc/fstab") > 0 &&
-           guestfs_is_file (g, "/etc/release") > 0) {
+           guestfs_is_file (g, "/etc/release") > 0 &&
+           guestfs_is_file (g, "/etc/redhat-release") == 0) {
     /* Ignore /dev/sda1 which is a shadow of the real root filesystem
      * that is probably /dev/sda5 (see:
      * http://www.freebsd.org/doc/handbook/disk-organization.html)
-- 
2.7.4

